src/main/java/org/jvnet/hudson/test/TemporaryDirectoryAllocator.java

```diff
@@ -116,13 +110,34 @@ public synchronized void disposeAsync() {
         new Thread("Disposing "+base) {
             @Override
             public void run() {
-                for (File dir : tbr)
+                for (File dir : tbr) {
+                    LOGGER.info(() -> "deleting " + dir);
                     try {
-                        new FilePath(dir).deleteRecursive();
-                    } catch (IOException | InterruptedException e) {
-                        e.printStackTrace();
+                        delete(dir.toPath());
+                    } catch (IOException e) {
+                        LOGGER.log(Level.WARNING, null, e);
                     }
+                }
             }
         }.start();
     }
+
+    private void delete(Path p) throws IOException {
+        LOGGER.fine(() -> "deleting " + p);
+        if (Files.isDirectory(p, LinkOption.NOFOLLOW_LINKS)) {
+            try (DirectoryStream<Path> children = Files.newDirectoryStream(p)) {
+                for (Path child : children) {
+                    delete(child);
+                }
+            }
+        }
+        try {
+            Files.deleteIfExists(p);
```

Comment:
```
[ERROR] hudson.LauncherTest.remotable  Time elapsed: 4.281 s  <<< ERROR!
java.nio.file.FileSystemException: C:\Jenkins\workspace\Core_jenkins_PR-8006\test\target\j h12573638483276859145\log: The process cannot access the file because it is being used by another process
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:275)
	at java.base/sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:110)
	at java.base/java.nio.file.Files.deleteIfExists(Files.java:1191)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:135)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.delete(TemporaryDirectoryAllocator.java:130)
	at org.jvnet.hudson.test.TemporaryDirectoryAllocator.dispose(TemporaryDirectoryAllocator.java:98)
	at org.jvnet.hudson.test.TestEnvironment.dispose(TestEnvironment.java:84)
	at …
```